src/java/picard/illumina/IlluminaBasecallsToFastq.java

```diff
@@ -487,7 +482,7 @@ public void encode(final FastqRecordsForCluster val) {
             if (numSampleBarcodes != val.sampleBarcodeRecords.length) throw new IllegalStateException();
             encodeArray(val.templateRecords);
             encodeArray(val.sampleBarcodeRecords);
- //           encodeArray(val.molecularBarcodeRecords);
+            encodeArray(val.molecularBarcodeRecords);
```

Comment:
I have no idea why it was uncommented to begin with! this was the reason the tool was failing in production. The tests didn't discover it because they were not spilling to disk and so encode wasn't called...I've changed that first (made the tests fail) and then fixed this.